.github/workflows/main.yml

```diff
@@ -0,0 +1,57 @@
+name: AM32-Firmware
+
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
+
+      - name: Install ARM SDK
+        run: make arm_sdk_install
+
+      - name: Get GitHub Build Number (ENV)
+        id: get_buildno
+        run: echo "GITHUBBUILDNUMBER=${{ github.run_number }}" >> $GITHUB_ENV
+        continue-on-error: true
+
+      - name: Get Pull Request Number
+        id: get_pullno
+        run: echo "PULL_NUMBER=$(echo "$GITHUB_REF" | awk -F / '{print $3}')" >> $GITHUB_ENV
+        if: startsWith(github.ref, 'refs/pull/')
+
+      - name: Make artifact name
+        id: make_artifactname
+        run: |
+          ARTIFACT_NAME="AM32-F051-${{ github.run_number }}"
```

Comment:
I like the naming - we should add the `AM32-F051-` prefix to the makefile.

Reply:
well, TBH this is the part I'm not sure about. If in this repo in future there will be support for F3 chips, the F051 name might causing misunderstanding. In general, I agree to change prefix in makefile.